### PR TITLE
[feature/sql-bool-builder] New syntax for DBAL query builder for boolean generation

### PR DESCRIPTION
**Pull request by brunoais**
Applying the new syntax as told by Nicofuma in IRC

https://tracker.phpbb.com/browse/PHPBB3-13652

### CODE REVIEW

**Comment by brunoais** [comment]
I still need to fix that documentation, though...

**Comment by Nicofuma** [comment]
That's fine now, but could you add some constants to avoid using magic numbers?

**Comment by brunoais** [comment]
OK. I'll add them when I get home.

**Comment by brunoais** [comment]
Issues happened... I'll add them today or tomorrow depending on how it goes.

**Comment by brunoais** [comment]
It had gone terribly wrong.
@Nicofuma I did the removal of the magic numbers today. Please check all is correct

File: phpBB/phpbb/db/driver/driver.php [hunk 1]
```diff
@@ -809,21 +818,21 @@ protected function _process_boolean_tree_first($operations_ary)
...
 		return $this->_process_boolean_tree($operations_ary) . "\n";
 	}
 
 	protected function _process_boolean_tree($operations_ary)
 	{
-		$operation = array_shift($operations_ary);
+		$operation = $operations_ary[self::LOGICAL_OP];
 
-		foreach ($operations_ary as &$condition)
+		foreach ($operations_ary[1] as &$condition)
```
Comment (Nicofuma) [inline phpBB/phpbb/db/driver/driver.php]: what is this `1`?

File: phpBB/phpbb/db/driver/driver.php [hunk 2]
```diff
@@ -844,40 +853,40 @@ protected function _process_boolean_tree($operations_ary)
...
 							// Typical 3 element clause with {left hand} {operator} {right hand}
-							switch ($condition[1])
+							switch ($condition[self::COMPARE_OP])
 							{
 								case 'IN':
 								case 'NOT_IN':
 
 									// As this is used with an IN, assume it is a set of elements for sql_in_set()
-									$condition = $this->sql_in_set($condition[0], $condition[2], $condition[1] === 'NOT_IN', true);
+									$condition = $this->sql_in_set($condition[self::LEFT_STMT], $condition[self::RIGHT_STMT], $condition[Cself::OMPARE_OP] === 'NOT_IN', true);
```
Comment (Nicofuma) [inline phpBB/phpbb/db/driver/driver.php]: You have a type in `Cself::OMPARE_OP`

**Comment by Nicofuma** [comment]
Otherwise looks good

File: phpBB/phpbb/db/driver/driver.php (hunk 2 above)
Comment (brunoais) [inline phpBB/phpbb/db/driver/driver.php]: Typo. Thanks for the warn.

File: phpBB/phpbb/db/driver/driver.php (hunk 1 above)
Comment (brunoais) [inline phpBB/phpbb/db/driver/driver.php]: Another typo.

**Comment by Nicofuma** [comment]
The tests are failing:
```
1) phpbb_boolean_processor_test::test_single_not_like
Invalid argument supplied for foreach()
/home/travis/build/phpbb/phpbb/phpBB/phpbb/db/driver/driver.php:833
/home/travis/build/phpbb/phpbb/phpBB/phpbb/db/driver/driver.php:826
/home/travis/build/phpbb/phpbb/phpBB/phpbb/db/driver/driver.php:790
/home/travis/build/phpbb/phpbb/tests/dbal/boolean_processor_test.php:38
```

```

**Comment by brunoais** [comment]
Sorry about all this delay. My weekends are being sooooo damn busy. Way more than usual.

**Comment by Nicofuma** [comment]
thanks for your contribution

**Comment by brunoais** [comment]
You're welcome!

**Pull request merged**